Makefile

```diff
@@ -49,33 +62,36 @@ build: $(SRCS_JS)
 
 build-test: $(SRCS_TEST_JS)
 
-build-rules:
-	@$(MAKE) -s -C $(TSLINT_RULES_DIR)
-
-clean-rules:
-	@$(MAKE) -s -C $(TSLINT_RULES_DIR) clean
+tslint: $(TSLINT_TARGET)
 
-tslint: build-rules $(TSLINT_TARGET)
-
-tslint-test: build-rules $(TSLINT_TEST_TARGET)
+tslint-test: $(TSLINT_TEST_TARGET)
 
 test-mocha: build build-test tslint-test
 	@$(MOCHA)
 
-clean: clean-rules
+# rules
+clean:
 	@$(RM) $(SRCS_JS) $(patsubst %.js,%.js.map,$(SRCS_JS))
 	@$(RM) $(SRCS_TEST_JS) $(patsubst %.js,%.js.map,$(SRCS_TEST_JS))
 	@$(RM) $(TSC_TARGET) $(TSLINT_TARGET) $(TSLINT_TEST_TARGET)
+	@$(RM) $(SRCS_TSLINT_RULES_JS)
+
 
-$(TSLINT_TARGET): $(SRCS_TS) $(TSLINT_CONFIG)
+$(TSLINT_TARGET): $(SRCS_TS) $(TSLINT_CONFIG) $(SRCS_TSLINT_RULES_JS)
 	@$(RM) $(TSLINT_TARGET)
-	@$(TSLINT) $(foreach file,$(SRCS_TS),-f $(file))
+	@$(TSLINT) -r $(TSLINT_RULES_DIR) $(foreach file,$(SRCS_TS),-f $(file))
 	@$(TOUCH) $(TSLINT_TARGET)
 
-$(TSLINT_TEST_TARGET): $(SRCS_TEST_TS) $(TSLINT_CONFIG)
+$(TSLINT_TEST_TARGET): $(SRCS_TEST_TS) $(TSLINT_CONFIG) $(SRCS_TSLINT_RULES_JS)
 	@$(RM) $(TSLINT_TEST_TARGET)
-	@$(TSLINT) $(foreach file,$(SRCS_TEST_TS),-f $(file))
+	@$(TSLINT) -r $(TSLINT_RULES_DIR) $(foreach file,$(SRCS_TEST_TS),-f $(file))
 	@$(TOUCH) $(TSLINT_TEST_TARGET)
 
-%.js: %.ts
-	@$(TSC) $<
+$(SRCS_TSLINT_RULES_JS): $(SRCS_TSLINT_RULES_TS)
+	@$(TSC) $(TSLINT_D) $<
```

Comment:
We can use $(SRCS_TSLINT_RULES_TS) instead of $< for consistency with the rules below.